tests/test_deprecated.py

```diff
@@ -75,24 +74,55 @@ class Pokemon(BaseModel):
         jp_name: str
 
     class PokemonList(BaseModel):
-        __root__: List[Pokemon]
+        root: List[Pokemon]
```

Comment:
Just to clarify, there's technically no reason why this couldn't still be called `__root__`, it's just that there's no reason it has to be called that anymore and we are in fact explicitly prohibiting `__root__` because we want to be able to give users an error so that they don't think that `__root__` models will just work with no changes?

Reply:
I'm not sure, maybe dunder/private attributes can't be used as field names - have you tried?

Reply:
see above `__root__` doesn't currently work.

Reply:
We are ignoring all dunder annotations when building fields out of the namespace; they essentially become class variables. (I'll note this is the same as the behavior in v1 for everything _except_ `__root__`.)

You are right that we are raising a special error for `__root__` to make sure users realize changes will be required.

We could make `__root__` work as a field name, and could even add these properties to get things as close to "just working" as possible. I'm not sure if that's the best idea given 1, Samuel has suggested we discourage this pattern going forward, and 2, if we try to make it "just work", people may expect us to fix any incompatibility bugs.

Reply:
That makes sense to me David and I agree. I do think the migration path should explain this reasoning, otherwise users might think that we arbitrarily are changing the field name in the guide when it's not needed and/or be surprised at it not working with `__root__`